### PR TITLE
Filter representations by latest version

### PR DESCRIPTION
**Pull request by iHiD**

### CODE REVIEW

File: app/commands/exercise/representation/search.rb [hunk 1]
```diff
@@ -4,15 +4,17 @@ class Exercise::Representation::Search
   # Use class method rather than constant for easier stubbing during testing
   def self.requests_per_page = 20
 
-  initialize_with mode: Mandate::NO_DEFAULT, mentor: nil, criteria: nil, track: Mandate::NO_DEFAULT,
-    order: :most_submissions, page: 1, paginated: true, sorted: true, only_mentored_solutions: false do
+  initialize_with mode: Mandate::NO_DEFAULT, representer_version: Mandate::NO_DEFAULT, track: Mandate::NO_DEFAULT,
+    mentor: nil, criteria: nil, only_mentored_solutions: false,
+    order: :most_submissions, page: 1, paginated: true, sorted: true do
```
Comment (iHiD) [inline app/commands/exercise/representation/search.rb:9]: Adds `representer_version`

File: db/schema.rb [hunk 2]
```diff
@@ -340,8 +340,15 @@
     t.text "draft_feedback_markdown"
     t.string "exercise_id_and_ast_digest_idx_cache"
     t.integer "num_published_solutions", limit: 2, default: 0, null: false
+    t.bigint "oldest_solution_id"
+    t.bigint "prestigious_solution_id"
```
Comment (iHiD) [inline db/schema.rb]:
```suggestion
```

File: app/assemblers/assemble_exercise_representations_admin.rb [hunk 3]
```diff
@@ -31,4 +32,5 @@ def representations
   end
 
   def track = Track.find_by(slug: params[:track_slug])
+  def representer_version = track.representations.maximum(:representer_version) || 1
```
Comment (iHiD) [inline app/assemblers/assemble_exercise_representations_admin.rb:35]: This is the only real change in these files. Adding this param.

File: app/commands/exercise/representation/search.rb [hunk 4]
```diff
@@ -4,15 +4,17 @@ class Exercise::Representation::Search
...
     @order = order.try(&:to_sym)
   end
 
   def call
     @representations = Exercise::Representation.
       includes(:exercise, :track).
-      where('num_submissions > 1')
+      where('num_submissions > 1').
+      where(representer_version:)
```
Comment (iHiD) [inline app/commands/exercise/representation/search.rb:17]: And then using it here.

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**